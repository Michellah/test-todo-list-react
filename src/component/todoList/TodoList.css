.bodytodo {
    display: flex;
    flex-direction: row;
    align-content: center;
    margin-top: 1em;
    font-family: Arial, Helvetica, sans-serif;
    height: 42vh;
}

.todo {
    flex-basis: 50%;
    margin: 0 1em;
    border-radius: 15px;
    border: solid 3px ;
}

.done {
    flex-basis: 50%;
    margin: 0 1em;
    border-radius: 15px;
    border: solid 3px ;
}

.title {
    margin: 0;
    padding: 8px;
}

.checkbox {
    text-align: left;
    margin: 5px;
    padding: 1em;
    color: rgb(23, 23, 23);
    border-radius: 25px;
    border: solid 2px rgb(112, 96, 96);
}

.inputAdd {
    border-radius: 25px;
    border: solid 2px rgb(112, 96, 96);
    padding: 1em;
    width: 30vw;
}
.inputAdd:focus {
  border: 2px solid #ccc;
  outline: none;
}
.scroller {
   overflow-y: scroll;
   height: 31vh;
   scrollbar-color: #8b8b89 #fbfbfb;
   scrollbar-width: thin;  
}

h2 {
    color: black;
}
