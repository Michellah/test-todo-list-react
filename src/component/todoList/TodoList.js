import { useState } from "react";
import './TodoList.css';

function TodoList() {
    const [inputValue, setInputValue] = useState("");
    const [todo, setTodo] = useState([])

    const createTodo = (title) => {
        return ({
            title: title,
            status: false
        });
    }


    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue != " ") {
            const newTodo = createTodo(inputValue);
            setTodo([...todo, newTodo])
            setInputValue("");
        }
    }

    const handleCheck = (index) => {
        const todoCopy = [...todo]
        const todoIndex = todoCopy[index]
        todoIndex.status = true;
        todoCopy[index] = todoIndex;

        setTodo([...todoCopy])
    }

    return (
        <>
            <div>
                <h1>My todo-list</h1>

                <form onSubmit={handleSubmit}>
                    <input
                        type="text"
                        name=""
                        id=""
                        value={inputValue}
                        onChange={(event) => setInputValue(event.target.value)}
                        placeholder="New todo..."
                        className="inputAdd"
                        data-testid="inputAdd"
                    />
                </form>
            </div>

            <div className="bodytodo">
                <div className="todo">
                    <h2 className="title">Todo</h2>
                    <hr />
                    <div className="scroller">
                        {todo.map((todo, index) => {
                            return (
                                !todo.status && <div className="checkbox"  data-testid="valueTodo">
                                    <input type="checkbox" name="" id="" onChange={() => handleCheck(index)} data-testid="checkbox"/>
                                    <span className="value">{todo.title}</span>
                                </div>
                            );
                        })}
                    </div>

                </div>

                <div className="done">
                    <h2 className="title">Done</h2>
                    <hr />
                    <div className="scroller" >{todo.map((todo) => {
                        return (
                            todo.status && <div className="checkbox" data-testid="valueDone">
                                
                                <span className="value" > {todo.title} </span>
                            </div>
                        );
                    })}</div>

                </div>
            </div>
        </>
    );
}

export default TodoList;