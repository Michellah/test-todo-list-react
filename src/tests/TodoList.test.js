import { cleanup, fireEvent, render, screen } from "@testing-library/react"
import renderer from "react-test-renderer";
import TodoList from "../component/todoList/TodoList";


afterEach(() => {
    cleanup();
})

it('should render the component onto the screen', () => {
    render(<TodoList />);
    expect(screen.getByTestId('inputAdd')).toBeInTheDocument();
});

it('renders the correct initial DOM', () => {
    const container = render(<TodoList />);
    const input = container.getByTestId('inputAdd');
    expect(input.getAttribute('value')).toBe('');
});

it('should render when a valid input is entered', () => {
    const inputValue = 'test'
    render(<TodoList />);
    const input = screen.getByTestId('inputAdd');
    fireEvent.change(input, { target: { value: inputValue } })
    expect(input).toBeEnabled();
});

it('should render when an invalid input is entered', () => {
    const inputValue = ' '
    render(<TodoList />);
    const input = screen.getByTestId('inputAdd');
    fireEvent.change(input, { target: { value: inputValue } })
    expect(input.value).toBe(' ');
});

it('it creates a new todo', () => {
    const container = render(<TodoList />);
    const fakeTodo = {
        title: 'test',
        status: false
    };
    const input = container.getByTestId('inputAdd');

    //Create the todo
    fireEvent.change(input, { target: { value: fakeTodo.title } });
    fireEvent.submit(input);


    const todo = container.getByTestId('valueTodo');
    // the name should be in the document
    expect(todo.textContent).toEqual('test');

    // the checkbox values is false
    const checkbox = container.queryByTestId('checkbox');
    expect(checkbox).not.toBeChecked;

    // the input field is blank
    expect(input.value).toBe('');

    // the todo should be in the document
    expect(todo).toBeInTheDocument;
});

it('it add to done', () => {
    const handleChange = jest.fn()

    const container = render(<TodoList />);

    const fakeTodo = {
        title: 'test',
        status: false
    };
    const input = container.getByTestId('inputAdd');

    fireEvent.change(input, { target: { value: fakeTodo.title } });
    fireEvent.submit(input);

    const todo = container.queryByTestId('valueTodo');

    const checkbox = container.queryByTestId('checkbox');
    expect(checkbox).toBeChecked;

    fakeTodo.status = true;

    expect(fakeTodo.status).toBe(true);
    
    
});

it('rend correctemen', () => {
    const container =  renderer
        .create(<TodoList/>)
        .toJSON();

    expect(container).toMatchSnapshot();    
})

// it('it add to done', () => {
//     const fakeTodo = {
//         title: 'test',
//         status: true
//     };
//     const container = render(<TodoList />);
//     const checkbox = container.queryByTestId('checkbox');
//     expect(checkbox).toBeChecked;


// });


// it('it adds in done', () => {
//     const container = render(<TodoList />);
//     const fakeTodo = {
//         title: 'test',
//         status: false
//     };
//     const input = container.getByTestId('inputAdd');

//     //Create the todo
//     fireEvent.change(input, { target: { value: fakeTodo.title } });
//     fireEvent.submit(input);


//     const todo = container.queryByTestId('valueTodo');

//     // the checkbox values is true
//     const checkbox = container.queryByTestId('checkbox');
//     expect(checkbox).toBeChecked;

//     const done = container.queryByTestId('valueDone');
//     expect(done).toEqual('test');

//     // the todo should be in the document
//     expect(todo).not.toBeInTheDocument;

//     expect(done).toBeInTheDocument;
// });
