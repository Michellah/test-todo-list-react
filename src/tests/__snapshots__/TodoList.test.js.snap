// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rend correctemen 1`] = `
Array [
  <div>
    <h1>
      My todo-list
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="inputAdd"
        data-testid="inputAdd"
        id=""
        name=""
        onChange={[Function]}
        placeholder="New todo..."
        type="text"
        value=""
      />
    </form>
  </div>,
  <div
    className="bodytodo"
  >
    <div
      className="todo"
    >
      <h2
        className="title"
      >
        Todo
      </h2>
      <hr />
      <div
        className="scroller"
      />
    </div>
    <div
      className="done"
    >
      <h2
        className="title"
      >
        Done
      </h2>
      <hr />
      <div
        className="scroller"
      />
    </div>
  </div>,
]
`;
